apps/teams-test-app/src/components/privateApis/MonetizationAPIs.tsx

```diff
@@ -24,8 +25,19 @@ const OpenPurchaseExperience = (): React.ReactElement =>
           throw new Error('planId and term are required on input, if provided');
         }
       },
-      submit: async planInfo => {
-        await monetization.openPurchaseExperience(planInfo);
+      submit: async (planInfo, setResult?) => {
+        if (getTestBackCompat()) {
```

Comment:
I'm not looking forward to having these `if (getTestBackCompat()` checks everywhere for every single `ApiWithoutInput`/`ApiWithTextInput`/etc. Is there something we can do at the common layer, in the react components themselves to make it cleaner?

One thought I have, is to update how `submit` prop on `ApiWithoutInput`/`ApiWithTextInput`/etc. is defined.

E.g. for `ApiWithoutInput` right now it's set as follows:

```
onClick: (setResult: (result: string) => void) => Promise<string>;
```

We could change it to allow either a single delegate (for APIs that don't have a callback equivalent) or an object with two delegates:

```
onClick:
  | ((setResult: (result: string) => void) => Promise<string>)
  | {
      withPromise: (setResult: (result: string) => void) => Promise<string>;
      withCallback: (setResult: (result: string) => void) => string;
    };
```

We then could have the component check which case we're in, and if it's the second version it could check which one to call:

```
onClick={async () => {
  if (typeof onClick === 'function') {
    setResult(await onClick(setResult));
  } else {
    if (getTestBackCompat()) {
      setResult(onClick.withCallback(setResult));
    } else {
      setResult(await onClick.withPromise(setResult));
    }
  }
}}
```

The same could be done for `ApiWithTextInput`:

```
  onClick:
    | ((input: Partial<T>) => Promise<string>)
    | {
        validateInput: (input: Partial<T>) => void;
        submit:
          | ((input: T, setResult: (result: string) => void) => Promise<string>)
          | {
              withPromise: (input: T, setResult: (result: string) => void) => Promise<string>;
              withCallback: (input: T, setResult: (result: string) => void) => string;
            };
      };
```

(the call site would be similar to the one without input)

With all that a single API that needs the v1.0 testing would be easier to define, something like this:

```
submit: {
  withPromise: async planInfo => {
    await monetization.openPurchaseExperience(planInfo);
    return 'monetization.openPurchaseExperience()' + noHostSdkMsg;
  },
  withCallback: (planInfo, setResult) => {
    const callback = (error: SdkError | null): void => {
      if (error) {
        setResult(JSON.stringify(error));
     } else {
        setResult('Success');
      }
    };
    monetization.openPurchaseExperience(callback, planInfo);
    return 'monetization.openPurchaseExperience()' + noHostSdkMsg;
  },
},
```

I'm curious if I'm the only one who would find this easier to read and reason about.

Also, this would allow us to do things like hide all the API boxes which do not implement the `withCallback` stuff when the query string is present.

Reply:
Great idea, I'll make that change now

Reply:
If we decide to go this route, we should update your other PR as well.

Reply:
Yup I made the changes on this PR, just testing it now and will update the other PR after. Thanks again for the assistance

Reply:
Created a separate PR for modifying the ApiWithoutInput call here https://github.com/OfficeDev/microsoft-teams-library-js/pull/804

Decided to go this route instead of modifying the existing PR because I foolishly made modifications to a few API calls that put `getTestBackCompat( )`  everywhere  and thought it would be better to pull out this shared logic into a separate PR and then make the changes to the old PR after these are reviewed/merged in

Side Note:  I think I will probably rename the function `getTestBackCompat( )` as well in a future PR to just `isBackCompat( )` or `isTeamsJSV1( )` to make it more obvious what it does.